text/0027-foreman-plugin-forge.md

```diff
@@ -0,0 +1,141 @@
+- Feature Name: A Forge for Foreman plugins
+- Start Date: 2016-10-24
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+A Forge for Foreman acting as a central resource for its growing plugin
+ecosystem.
+
+# Motivation
+[motivation]: #motivation
+
+Foreman has an expanding plugin ecosystem that includes plugins for compute
+resources, config management integrations and provisioning. However,
+discoverability experience can be made better.
+
+There's a page for [Plugins](https://theforeman.org/plugins) on the Foreman
+website,
+a
+[list of plugins](http://projects.theforeman.org/projects/foreman/wiki/List_of_Plugins) on
+the wiki, apart from various official/user Github pages for further details
+about each plugin. Documentation for plugins are also available at multiple
+places. For example, [Foreman remote execution](https://theforeman.org/plugins/foreman_remote_execution/0.3/index.html) plugin's
+docs are a part of the official website, but many plugins maintain docs within
+README.md file in the project root.
+
+Keeping information up to date on each page is a manual task and could use some
+automation.
+
+A new Forge would address the following points:
+
+* Showcasing Foreman plugin ecosystem and discover new plugins
+* Offer helpful commands/code snippet for installing the plugin via source, gem,
+  rpm, deb
+* Display a short README à la Github's rendering of README.md
+* Show plugin popularity/trustworthiness via download stats
+* Plugin management (list/install/uninstall) via a command like
+  foreman-rake/hammer - does it even make sense for us to do?
+
+# Detailed design
+[design]: #detailed-design
+
+A Foreman Forge page for plugin would provide all the information about that
```

Comment:
+1 For compatibility matrix. I also think a list of versions with release notes (and compatibility matrix for each of them) would be handy too.